internal/db/data_flow.md

```diff
@@ -0,0 +1,544 @@
+
+### Storage Components
+
+**Implementation Files:**
+- `datastore/store.go` - MultiStore interface definition
+- `datastore/blockstore.go` - Blockstore wrapper
+- `datastore/txn.go` - Transaction handling
+- `internal/keys/` - Key formatting for each store type
+
+The `MultiStore` interface defines all available stores:
+
+```go
+type MultiStore interface {
+    Rootstore()   DSReaderWriter  // Base transactional store
+    Datastore()   DSReaderWriter  // Document data (/data namespace)
+    Encstore()    Blockstore      // Encryption keys (/enc namespace)
+    Headstore()   DSReaderWriter  // Document heads (/head namespace)
+    Peerstore()   DSReaderWriter  // Peer information (/peers namespace)
+    Blockstore()  Blockstore      // IPLD blocks (/blocks namespace)
+    Systemstore() DSReaderWriter  // System metadata (/system namespace)
+}
+```
+
+### AddDelta Function Flow
+
+**Related Files:**
+- `internal/core/block/store.go` - AddDelta implementation and block storage
+- `internal/core/block/block.go` - Block structure definitions
+- `internal/core/block/encryption.go` - Encryption block handling
+- `internal/core/block/signature.go` - Block signing
+- `internal/encryption/encryptor.go` - Encryption implementation
+
+The `AddDelta` function is called by the collection's save method after creating CRDTs:
+- **For field updates**: Called once per modified field with the field's delta
+- **For composite blocks**: Called with the composite block containing all field references
+
+**When AddDelta is invoked:**
+1. During document creation (`collection.create()`)
+2. During document updates (`collection.update()`)
+3. Both operations call `collection.save()`, which creates CRDTs and calls AddDelta
+
+The `AddDelta` function orchestrates the storage process:
+
+1. **Height Update**: Increments the block height for version tracking
+2. **Encryption Check**: 
+   - If encryption is enabled (`internal/core/block/encryption.go`):
+   - Creates encryption block with decryption key
+   - Attaches as IPLD link to main block
+   - Encrypts block content (`internal/encryption/encryptor.go`)
+   - Replaces plaintext with ciphertext
+3. **Signing**: If enabled, signs the block for integrity verification
+4. **Storage**:
+   - **Encrypted/signed block → Blockstore**: The transformed block (with encryption and/or signature) is stored in the blockstore. This is the version that gets distributed to other peers during synchronization.
+   - **Original block → Local processing**: The unencrypted block data is used for local operations like updating indexes and merging CRDT states. This ensures the local node can work with the data without decryption overhead.
+5. **Local Processing**:
+   - Merges CRDT changes into the current document state
+   - Updates document heads in headstore to track the latest version
+   - Triggers index updates for query optimization
```

Comment:
suggestion: It would be worth adding a point here (under `5.`) to note that this is where stuff is written to the datastore too, it feels like an important omission.